sw/device/tests/spi_device_sleep_test.c

```diff
@@ -0,0 +1,157 @@
+// Copyright lowRISC contributors (OpenTitan project).
+// Licensed under the Apache License, Version 2.0, see LICENSE for details.
+// SPDX-License-Identifier: Apache-2.0
+
+#include "sw/device/lib/base/mmio.h"
+#include "sw/device/lib/dif/dif_aon_timer.h"
+#include "sw/device/lib/dif/dif_pinmux.h"
+#include "sw/device/lib/dif/dif_pwrmgr.h"
+#include "sw/device/lib/dif/dif_rv_core_ibex.h"
+#include "sw/device/lib/dif/dif_rv_plic.h"
+#include "sw/device/lib/runtime/irq.h"
+#include "sw/device/lib/runtime/log.h"
+#include "sw/device/lib/testing/aon_timer_testutils.h"
+#include "sw/device/lib/testing/pwrmgr_testutils.h"
+#include "sw/device/lib/testing/rv_plic_testutils.h"
+#include "sw/device/lib/testing/spi_device_testutils.h"
+#include "sw/device/lib/testing/spi_flash_emulator.h"
+#include "sw/device/lib/testing/spi_flash_testutils.h"
+#include "sw/device/lib/testing/test_framework/check.h"
+#include "sw/device/lib/testing/test_framework/ottf_main.h"
+
+#include "hw/top_earlgrey/sw/autogen/top_earlgrey.h"
+#include "sw/device/lib/testing/autogen/isr_testutils.h"
+
+OTTF_DEFINE_TEST_CONFIG();
+
+enum {
+  kWakeupTimeMicros = 500000,
+};
+static dif_spi_device_handle_t spid;
+static dif_pwrmgr_t pwrmgr;
+static dif_aon_timer_t aon_timer;
+static dif_rv_plic_t plic;
+static plic_isr_ctx_t plic_ctx = {.rv_plic = &plic,
+                                  .hart_id = kTopEarlgreyPlicTargetIbex0};
+static pwrmgr_isr_ctx_t pwrmgr_isr_ctx = {
+    .pwrmgr = &pwrmgr,
+    .plic_pwrmgr_start_irq_id = kTopEarlgreyPlicIrqIdPwrmgrAonWakeup,
+    .expected_irq = kDifPwrmgrIrqWakeup,
+    .is_only_irq = true};
+
+static void enter_low_power(void) {
+  uint64_t wkup_cycles = 0;
+  CHECK_STATUS_OK(aon_timer_testutils_get_aon_cycles_64_from_us(
+      kWakeupTimeMicros, &wkup_cycles));
+  CHECK_STATUS_OK(aon_timer_testutils_wakeup_config(&aon_timer, wkup_cycles));
+
+  dif_pwrmgr_domain_config_t pwrmgr_domain_cfg =
+      kDifPwrmgrDomainOptionMainPowerInLowPower;
```

Comment:
Should we be testing this for deep sleep?

@jesultra @jettr Would you have some idea of whether spi_device's responses in deep sleep are relevant?

Reply:
We do not have a requirement for the spi_device to respond during deep sleep. We do have a requirement that activity on the SPI device should wake the device up from normal sleep, but it is not an expectation to capture the incoming SPI transaction while waking -- the spi host will resend the data packet if needed.